packages/arb-token-bridge-ui/src/components/common/Dialog.tsx

```diff
@@ -84,84 +83,123 @@ export function useDialog(params?: UseDialogParams): UseDialogResult {
 
 export type DialogProps = {
   isOpen: boolean
-  isCustom?: boolean
   closeable?: boolean
   title?: string | JSX.Element
   initialFocus?: React.MutableRefObject<HTMLElement | null>
   cancelButtonProps?: Partial<ButtonProps>
   actionButtonProps?: Partial<ButtonProps>
   actionButtonTitle?: string
+  isFooterHidden?: boolean
   onClose: (confirmed: boolean, onCloseData?: unknown) => void
   className?: string
   children?: React.ReactNode
 }
 
 export function Dialog(props: DialogProps) {
-  const isCustom = props.isCustom || false
-  const closeable = props.closeable || false
+  const isFooterHidden = props.isFooterHidden || false
+  const closeable = props.closeable ?? true
   const className = props.className || ''
   const cancelButtonRef = useRef(null)
 
+  // separate state to track transition state and have a smooth exit animation
+  const [isClosing, setIsClosing] = useState(false)
+  const [isConfirmed, setIsConfirmed] = useState(false)
+
+  const handleCloseStart = useCallback(
+    (confirmed: boolean) => {
+      if (!confirmed && !closeable) {
+        return
+      }
+
+      setIsConfirmed(confirmed)
+      setIsClosing(true)
+    },
+    [setIsClosing, setIsConfirmed, closeable]
+  )
+
+  const handleCloseEnd = useCallback(() => {
+    props.onClose(isConfirmed)
+    // prevents race conditions that could cause a flicker of the dialog after close
+    setTimeout(() => {
+      setIsClosing(false)
+    }, 0)
+  }, [props, isConfirmed, setIsClosing])
+
   return (
-    <Transition
-      appear
-      show={props.isOpen}
-      as={Fragment}
-      {...getTransitionProps('normal')}
-    >
+    <Transition show={props.isOpen && !isClosing} as={Fragment}>
```

Comment:
Transition was redone because the old transition didn't work anymore. 

For the transition to work the initial opened state needs to be false. But because we rendered some of our dialogs conditionally, the initial internal opened state would always be true, even if the component was conditionally hidden.

In the old implementation the background also didn't work. Its opacity didn't smoothly transition, and it wasn't clickable to close the dialog.